src/main/java/com/mcmoddev/mmdbot/modules/commands/info/CmdTranslateMappings.kt

```diff
@@ -0,0 +1,205 @@
+                                    .addField("$namespace2 Name", "`${translation.optimumName}`", false)
+                            }
+                            else -> {
+                                EmbedBuilder().setDescription("???")
+                            }
+                        }.setFooter("Page ${idx + 1} | Powered by linkie-core").build()
+                }.iterator()
+
+            if (!embeds.hasNext()) {
+                embeds = listOf(EmbedBuilder()
+                    .setTitle("$namespace1 -> $namespace2 Class mapping for $version:")
+                        .setDescription("No results found.")
+                    .setFooter("Powered by linkie-core")
+                    .setColor(Color.RED)
+                    .build()
+                ).iterator()
+            }
+
+            val msg = event.channel.sendMessageEmbeds(embeds.next()).apply {
+                if (embeds.hasNext()) {
+                    setActionRow(Button.primary("mappings-trans-next", "Next"))
+                }
+            }.complete()
+
+            ButtonListener.embedsForMessage[msg.idLong] = embeds
+
+            delay(180000L)
+
+            ButtonListener.embedsForMessage.remove(msg.idLong)
+            event.channel.editMessageById(msg.id, msg).setActionRows().complete()
+        }
+    }
+
+    /**
+     * Translates a mapped class, field, or member, [original], to the [target] mappings.
+     *
+     * If [original] is a `ResultHolder<T>`, then the method will return `T?`. Unfortunately this cannot be expressed
+     * in the code.
+     *
+     * @param original the thing to be translated.
+     * @param target the target mappings.
+     *
+     * @return the translated [Class], [Field], or [Method], or null if none was found.
+     */
+    private fun translate(original: ResultHolder<*>, target: MappingsContainer): Any? {
+        return when {
+            original.value is Class -> {
+                (original.value as Class).obfMergedName?.let { target.getClassByObfName(it) }
+            }
+            original.value is Pair<*, *> && (original.value as Pair<*, *>).second is Field -> {
+                val parent = (original.value as Pair<*, *>).first as Class
+                val member = (original.value as Pair<*, *>).second as Field
+
+                member.obfMergedName?.let { memberName -> parent.obfMergedName?.let { className -> target.getClassByObfName(className)?.getFieldByObfName(memberName) } }
+            }
+            original.value is Pair<*, *> && (original.value as Pair<*, *>).second is Method -> {
+                val parent = (original.value as Pair<*, *>).first as Class
+                val member = (original.value as Pair<*, *>).second as Method
+
+                member.obfMergedName?.let { memberName -> parent.obfMergedName?.let { className -> target.getClassByObfName(className)?.getMethodByObfName(memberName) } }
+            }
```

Comment:
You might want to store `original.value` in a `val` to leverage smart casts and get rid of some of the noise

Reply:
Thanks, fixed.